File: docs/vsto/codesnippet/CSharp/trin_wordcontentcontrolreference/DropDownList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Word = Microsoft.Office.Interop.Word;
using Office = Microsoft.Office.Core;

namespace Trin_ContentControlReference
{
    public partial class ThisDocument
    {

        // DropDownListContentControl constructor #1: create a DropDownListContentControl 
        // at the current selection.
        //<Snippet300>
        private Microsoft.Office.Tools.Word.DropDownListContentControl dropDownListControl1;

        private void AddDropDownListControlAtSelection()
        {
            this.Paragraphs[1].Range.InsertParagraphBefore();
            this.Paragraphs[1].Range.Select();

            dropDownListControl1 = this.Controls.AddDropDownListContentControl("dropDownListControl1");
            dropDownListControl1.DropDownListEntries.Add("Monday", "Monday", 0);
            dropDownListControl1.DropDownListEntries.Add("Tuesday", "Tuesday", 1);
            dropDownListControl1.DropDownListEntries.Add("Wednesday", "Wednesday", 2);
            dropDownListControl1.PlaceholderText = "Choose a day";
        }
        //</Snippet300>

        // DropDownListContentControl constructor #2: create a DropDownListContentControl at a specified range.
        //<Snippet301>
        private Microsoft.Office.Tools.Word.DropDownListContentControl dropDownListControl2;

        private void AddDropDownListControlAtRange()
        {
            this.Paragraphs[1].Range.InsertParagraphBefore();

            dropDownListControl2 = this.Controls.AddDropDownListContentControl(this.Paragraphs[1].Range,
                "dropDownListControl2");
            dropDownListControl2.DropDownListEntries.Add("Monday", "Monday", 0);
            dropDownListControl2.DropDownListEntries.Add("Tuesday", "Tuesday", 1);
            dropDownListControl2.DropDownListEntries.Add("Wednesday", "Wednesday", 2);
            dropDownListControl2.PlaceholderText = "Choose a day";
        }
        //</Snippet301>

        // DropDownListContentControl constructor #3: create a DropDownListContentControl for every
        // native drop down list control in the document.
        //<Snippet302>
        private System.Collections.Generic.List
            <Microsoft.Office.Tools.Word.DropDownListContentControl> dropDownControls;

        private void CreateDropDownListControlsFromNativeControls()
        {
            if (this.ContentControls.Count <= 0)
                return;

            dropDownControls = new System.Collections.Generic.List
                <Microsoft.Office.Tools.Word.DropDownListContentControl>();
            int count = 0;

            foreach (Word.ContentControl nativeControl in this.ContentControls)
            {
                if (nativeControl.Type == Word.WdContentControlType.wdContentControlDropdownList)
                {
                    count++;
                    Microsoft.Office.Tools.Word.DropDownListContentControl tempControl =
                        this.Controls.AddDropDownListContentControl(nativeControl,
                        "VSTODropDownListContentControl" + count.ToString());
                    dropDownControls.Add(tempControl);
                }
            }
        }
        //</Snippet302>


        // DropDownListContentControl constructor #3: create a DropDownListContentControl for every
        // native drop down list control added to the document.
        //<Snippet303>
        void ThisDocument_DropDownListContentControlAfterAdd(Word.ContentControl NewContentControl, bool InUndoRedo)
        {
            if (NewContentControl.Type == Word.WdContentControlType.wdContentControlDropdownList)
            {
                this.Controls.AddDropDownListContentControl(NewContentControl,
                    "DropDownListControl" + NewContentControl.ID);
            }
        }
        //</Snippet303>
    }
}

File: docs/vsto/codesnippet/VisualBasic/trin_contentcontrolreference/DropDownList.vb
Partial Public NotInheritable Class ThisDocument


    ' DropDownListContentControl constructor #1: create a DropDownListContentControl at the current selection.
    '<Snippet300>
    Dim dropDownListControl1 As Microsoft.Office.Tools.Word.DropDownListContentControl

    Private Sub AddDropDownListControlAtSelection()
        Me.Paragraphs(1).Range.InsertParagraphBefore()
        Me.Paragraphs(1).Range.Select()
        dropDownListControl1 = Me.Controls.AddDropDownListContentControl("dropDownListControl1")
        With dropDownListControl1
            .DropDownListEntries.Add("Monday", "Monday", 0)
            .DropDownListEntries.Add("Tuesday", "Tuesday", 1)
            .DropDownListEntries.Add("Wednesday", "Wednesday", 2)
            .PlaceholderText = "Choose a day"
        End With
    End Sub
    '</Snippet300>

    ' DropDownListContentControl constructor #2: create a DropDownListContentControl at a specified range.
    '<Snippet301>
    Dim dropDownListControl2 As Microsoft.Office.Tools.Word.DropDownListContentControl

    Private Sub AddDropDownListControlAtRange()
        Me.Paragraphs(1).Range.InsertParagraphBefore()
        dropDownListControl2 = Me.Controls.AddDropDownListContentControl(Me.Paragraphs(1).Range, _
            "dropDownListControl2")
        With dropDownListControl2
            .DropDownListEntries.Add("Monday", "Monday", 0)
            .DropDownListEntries.Add("Tuesday", "Tuesday", 1)
            .DropDownListEntries.Add("Wednesday", "Wednesday", 2)
            .PlaceholderText = "Choose a day"
        End With
    End Sub
    '</Snippet301>

    ' DropDownListContentControl constructor #3: create a DropDownListContentControl for every
    ' native drop down list control in the document.
    '<Snippet302>
    Private dropDownListControls As New System.Collections.Generic.List _
            (Of Microsoft.Office.Tools.Word.DropDownListContentControl)

    Private Sub CreateDropDownListControlsFromNativeControls()
        If Me.ContentControls.Count <= 0 Then
            Return
        End If

        Dim count As Integer = 0
        For Each nativeControl As Word.ContentControl In Me.ContentControls
            If nativeControl.Type = Word.WdContentControlType.wdContentControlDropdownList Then
                count += 1
                Dim tempControl As Microsoft.Office.Tools.Word.DropDownListContentControl = _
                    Me.Controls.AddDropDownListContentControl(nativeControl, _
                    "VSTODropDownListContentControl" + count.ToString())
                dropDownListControls.Add(tempControl)
            End If
        Next nativeControl
    End Sub
    '</Snippet302>


    ' DropDownListContentControl constructor #3: create a 
    ' DropDownListContentControl for every native drop down list added to the document.
    '<Snippet303>
    Private Sub ThisDocument_DropDownListContentControlAfterAdd(ByVal NewContentControl As Word.ContentControl, _
        ByVal InUndoRedo As Boolean) Handles Me.ContentControlAfterAdd

        If NewContentControl.Type = Word.WdContentControlType.wdContentControlDropdownList Then
            Me.Controls.AddDropDownListContentControl(NewContentControl, _
                "DropDownListControl" + NewContentControl.ID)
        End If
    End Sub
    '</Snippet303>

End Class
